config/src/main/java/com/typesafe/config/impl/ConfigDocumentParser.java

```diff
@@ -197,7 +211,7 @@ private ConfigException parseError(String message) {
         }
 
         private ConfigException parseError(String message, Throwable cause) {
-            return new ConfigException.Parse(SimpleConfigOrigin.newSimple(""), message, cause);
+            return new ConfigException.Parse(SimpleConfigOrigin.newSimple("").withLineNumber(lineNumber), message, cause);
```

Comment:
the old parser has a `baseOrigin` passed in, rather than making up this `newSimple("")` origin - I would think ConfigDocumentParser.parse should take an origin too? then you don't have to make one up here. If the `newSimple("")` origin escapes into public API, then we probably need a test that it doesn't... we shouldn't ever be returning an empty/useless origin like that.